Forge.TreeWalker/src/TreeSchemaValidator.cs

```diff
@@ -0,0 +1,275 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="TreeSchemaValidator.cs" company="Microsoft">
+//     Copyright (c) Microsoft Corporation.  All rights reserved.
+// </copyright>
+// <summary>
+//     The TreeSchemaValidator class implements the ITreeSchemaValidator interface.
+// </summary>
+//-----------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.Serialization.Json;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Forge.DataContracts;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Newtonsoft.Json.Schema;
+
+
+namespace Microsoft.Forge.TreeWalker
+{
+    public class TreeSchemaValidator : ITreeSchemaValidator
+    {
+        private JSchema ForgeSchemaValidationRules;
+        public TreeSchemaValidator() => ForgeSchemaValidationRules = GetRulesFromPath("contracts\\ForgeSchemaValidationRules.json");
+
+        public async Task<Tuple<bool, IList<string>>> ValidateSchema(ForgeTree schema)
```

Comment:
>public async Task<Tuple<bool, IList<string>>> [](http://example.com/codeflow?start=8&length=45)

I don't think we're doing anything asynchronous. Let's remove all async and Tasks from this class.

If folks wanted to kick off validating their schemas, they could wrap the method call in a Task themselves. Though validation shouldn't take any significant time, so this should be all be fine running synchronously.